Controller/Adminhtml/Importcategory/Save.php

```diff
@@ -80,138 +80,165 @@ public function execute()
-                                            $cateitem->setParentId($_parentid);
-                                            $cateitem->setPath($parentcategory->getPath());
+
+                                    if ($_parentId!='' && $newCategory!='') {
+                                        $cateItem = $this->_categoryFactory->create();
+                                        $cateItem->setData($cat_data);
+                                        $parentCategory = $this->_categoryFactory->create();
+                                        $parentCategory->load($_parentId);
+
+                                        if ($parentCategory->getId()) {
+                                            $cateItem->setParentId($_parentId);
+                                            $cateItem->setPath($parentCategory->getPath());
                                         }
-                                        $cateitem->setAttributeSetId($cateitem->getDefaultAttributeSetId());
-                                        $cateitem->setName($newcategory);
 
-                                        $_url_key =  $cat_data['url_key'] ? $cat_data['url_key'] : str_replace(' ', '-', strtolower($newcategory));
-                                        if (in_array($newcategory, $exist_categories_name)) {
+                                        $cateItem->setAttributeSetId($cateItem->getDefaultAttributeSetId());
+                                        $cateItem->setName($newCategory);
+
+                                        // if url_key is specified use that key, otherwise use generated key
+                                        $_url_key =  $cat_data['url_key'] ? $cat_data['url_key'] : str_replace(' ', '-', strtolower($newCategory));
+
+                                        if (in_array($newCategory, $existCategoriesName)) {
                                             $_url_key .= '-'.mt_rand(10, 99);
                                         }
-                                        $cateitem->setUrlKey($_url_key);
+                                        $cateItem->setUrlKey($_url_key);
 
-                                        $cateitem->setStoreId($storeId);
-                                        $cateitem->save();
-                                        if ($cateitem->getId()) {
-                                            $exist_categories_name[$cateitem->getId()] = $cateitem->getName();
-                                            $exist_categories_path[$cateitem->getId()] = $cateitem->getPath();
-                                            $exist_categories_pathname[$cateitem->getId()] = $categoryitem[$categorieskey];
+                                        $cateItem->setStoreId($storeId);
+                                        $cateItem->save();
+                                        if ($cateItem->getId()) {
+                                            $existCategoriesName[$cateItem->getId()] = $cateItem->getName();
```

Comment:
please rename this variable 

be more descriptive

`$existingCategories` maybe